### DISCUSSION TITLE
Error when listing all files in user_home disk

### DISCUSSION THREAD

**Discussion by danielpetrica** [NativePHP/laravel#374] [Q&A] [answered]
Hi I encoundered a strange issue which I don't know if it comes from nativephp or Laravel. 

when doing` Storage::disk('user_home')->allFiles(); ` I get an error relative to deep linking 

<details><summary>Error message: </summary>
<p>

League\Flysystem\UnableToListContents
Unable to list contents for '', deep listing Reason: Unsupported symbolic link encountered at location /Users/daniel/.config/raycast/extensions/node_modules
POST 127.0.0.1:8100
PHP 8.3.11 — Laravel 11.23.5

</p>
</details> 

Doing a ls -la command I can see that the node_modules listed there is indeed a symlink
<details><summary>Ls output</summary>
<p>

```
lrwxr-xr-x@  1 daniel  staff   131 Sep 24 14:20 node_modules -> /Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules
```
</p>
</details> 

I know the exception is triggered because it meets a symlink, but could I patch it in any way? Is it something you encountered before? 

I'd like to avid writing a function to list all files recursively by hand.

**Comment by danielpetrica** [comment]
OK I found a solution. I'll try to push a pr for that but I need some feedback on how to proceed. 
Basically flysystem doesn't support symlinks  ( https://github.com/thephpleague/flysystem/issues/599 ). But I found a workaround here https://github.com/UniSharp/laravel-filemanager/issues/723#issuecomment-930668029  which consists in adding a simple  `'links' => 'skip',` to the configuration. 
I'll try to edit the NativeServiceProvider.php to add the skip symlinks option. 
Would that be ok with the package?

**Reply by simonhamp** [reply]
This seems reasonable to me 👍🏼

**Reply by simonhamp** [reply]
[Linking the PR](https://github.com/NativePHP/laravel/pull/378)

**Reply by simonhamp** [reply, marked as answer]
This was released, so should be resolved now.
